### ISSUE TITLE
[Bug]: The cast conversion problem when the format() function parameter is double type

### ISSUE THREAD

**Issue by Ariznawlll** [matrixorigin/matrixone#7356]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

commit 914bab62256ee1f109cd33601cd4cbf2010b735c
./mo-service -launch ./etc/launch-tae-CN-tae-DN/launch.toml
mysql -h 127.0.0.1 -P 6001 -udump -p111

CREATE TABLE t(id INT, d1 FLOAT, d2 DOUBLE NOT NULL);

INSERT INTO t VALUES(4, 0, -38902093.4324);
INSERT INTO t VALUES(5, 323, 37827329.43432);
INSERT INTO t VALUES(8, 7382,3923404.2132421);

SELECT format(d2, id) from t;

mo result:-3.8902 
Mysql result:-38,902,093.4324

**Comment by YANGGMM** [comment]
this is a  cast error, cast double as varchar will lead to this error

**Comment by iceTTTT** [comment]
Should be fixed by pr #7398 .
